### PR TITLE
Prevent drawing overlay in specific cases (such as movie mode, etc.)

### PR DESCRIPTION
**Pull request by RattleSN4K3**

### CODE REVIEW

**Comment by WGH-** [comment]
Wow, do you really still play the game? :)

File: UTGame/Src/SpectatorUI/Classes/SpectatorUI_Interaction.uc [hunk 1]
```diff
@@ -118,8 +118,31 @@ function Cleanup() {
...
 
 function bool ShouldRender() {
-    // the same condition appears in UTHUD::DrawGameHud
-    return PlayerReplicationInfo != None && (PlayerReplicationInfo.bOnlySpectator || Outer.IsInState('Spectating'));
+	local GameReplicationInfo GRI;
+	local UTGameReplicationInfo UTGRI;
+	local UTHUD HUD;
+	
+	// the same condition appears in UTHUD::DrawGameHud
+	if (PlayerReplicationInfo == none || !PlayerReplicationInfo.bOnlySpectator || !Outer.IsInState('Spectating'))
```
Comment (WGH-) [inline UTGame/Src/SpectatorUI/Classes/SpectatorUI_Interaction.uc]: That's not the negation of the original condition. It should be `PlayerReplicationInfo == none || !PlayerReplicationInfo.bOnlySpectator && !Outer.IsInState('Spectating')` instead, no?

**Comment by RattleSN4K3** [comment]
> Wow, do you really still play the game? :)

Not really. Playedthe last time about 3 years ago. But occasionally I keep finishing some stuff I did. However, this was actually an request by someone who wrote a PM directed at me. I suggested to create a thread for it ([so he did](https://www.epicgames.com/unrealtournament/forums/showthread.php?23650-Request-SpectatorUI-v2-5-adaption)).

> That's not the negation of the original condition. It should be `PlayerReplicationInfo == none || !PlayerReplicationInfo.bOnlySpectator && !Outer.IsInState('Spectating')` instead, no?

Seems like it isn't the right negation but from the condition, I think `PlayerReplicationInfo == none || !PlayerReplicationInfo.bOnlySpectator` is enough. In words: _only render if there's a player and the player is a spec_.

Currently (with the PR applied), the overlay might not be drawn if there's a spectator which isn't in the `Spectating` state. Not sure if the demo spectator could be in such condition. But I feel like this isn't required at all, this check.

By judging the original conditions, should the overlay be drawn if the player dies and waits for the respawn (or in a round game and waiting for the next round)?

**Comment by WGH-** [comment]
I don't remember why I written the condition that way. It might have something to do with game modes where players may become spectators without being in "spectator only" mode, like Duel (when waiting in the queue). I tested many different games modes, including some third-party ones, like Jailbreak. 

Why can't you just leave the condition as it originally is? :)

**Comment by WGH-** [comment]
> By judging the original conditions, should the overlay be drawn if the player dies and waits for the respawn?

I don't think so, as it would be `Dead` state.

**Comment by RattleSN4K3** [comment]
> Why can't you just leave the condition as it originally is? :)

I corrected/reverted it.

**Pull request merged**